File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
function isEven(n) { return n % 2 === 0; }

function helper(n, acc) {
  if (n === 1) return acc;
  return isEven(n) ? helper(n / 2, acc + 1) : helper(3 * n + 1, acc + 1);
}

export const steps = (n) => {
  if (n <= 0) throw new Error("Only positive integers are allowed");
  return helper(n, 0);
};

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int helper(int start, int count) {
    return start == 1? count :
           start &  1? helper(3 * start + 1, count + 1) :
           helper(start >> 1, count + 1);
}

int steps(int start) {
    return start <  1? ERROR_VALUE : helper(start, 0);
}